enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/BuildEnterpriseInterpretedExecutionPlan.scala

```diff
@@ -47,29 +48,33 @@ object BuildEnterpriseInterpretedExecutionPlan extends Phase[EnterpriseRuntimeCo
-    val idMap = LogicalPlanIdentificationBuilder(logicalPlan)
-    val converters = new ExpressionConverters(CommunityExpressionConverter, EnterpriseExpressionConverters)
-    val executionPlanBuilder = new PipeExecutionPlanBuilder(context.clock, context.monitors,
-      expressionConverters = converters, pipeBuilderFactory = RegisteredPipeBuilderFactory(pipelines))
-    val pipeBuildContext = PipeExecutionBuilderContext(context.metrics.cardinality, from.semanticTable(), from.plannerName)
-    val pipeInfo = executionPlanBuilder.build(from.periodicCommit, logicalPlan, idMap)(pipeBuildContext, context.planContext)
-    val PipeInfo(pipe: Pipe, updating, periodicCommitInfo, fp, planner) = pipeInfo
-    val columns = from.statement().returnColumns
-    val resultBuilderFactory = DefaultExecutionResultBuilderFactory(pipeInfo, columns, logicalPlan, idMap)
-    val func = BuildInterpretedExecutionPlan.getExecutionPlanFunction(periodicCommitInfo, from.queryText, updating, resultBuilderFactory, context.notificationLogger)
-    val fingerprint = context.createFingerprintReference(fp)
-    val periodicCommig = periodicCommitInfo.isDefined
-    val indexes = logicalPlan.indexUsage
-    val execPlan = RegisteredExecutionPlan(fingerprint, periodicCommig, planner, indexes, func, pipe, context.config)
-    new CompilationState(from, Some(execPlan))
+    val runtimeSuccessRateMonitor = context.monitors.newMonitor[NewRuntimeSuccessRateMonitor]()
+    try {
+      val (logicalPlan, pipelines) = rewritePlan(context, from.logicalPlan)
+      val idMap = LogicalPlanIdentificationBuilder(logicalPlan)
+      val converters = new ExpressionConverters(CommunityExpressionConverter, EnterpriseExpressionConverters)
+      val executionPlanBuilder = new PipeExecutionPlanBuilder(context.clock, context.monitors,
+        expressionConverters = converters, pipeBuilderFactory = RegisteredPipeBuilderFactory(pipelines))
+      val pipeBuildContext = PipeExecutionBuilderContext(context.metrics.cardinality, from.semanticTable(), from.plannerName)
+      val pipeInfo = executionPlanBuilder.build(from.periodicCommit, logicalPlan, idMap)(pipeBuildContext, context.planContext)
+      val PipeInfo(pipe: Pipe, updating, periodicCommitInfo, fp, planner) = pipeInfo
+      val columns = from.statement().returnColumns
+      val resultBuilderFactory = DefaultExecutionResultBuilderFactory(pipeInfo, columns, logicalPlan, idMap)
+      val func = BuildInterpretedExecutionPlan.getExecutionPlanFunction(periodicCommitInfo, from.queryText, updating, resultBuilderFactory, context.notificationLogger)
+      val fingerprint = context.createFingerprintReference(fp)
+      val periodicCommig = periodicCommitInfo.isDefined
+      val indexes = logicalPlan.indexUsage
+      val execPlan = RegisteredExecutionPlan(fingerprint, periodicCommig, planner, indexes, func, pipe, context.config)
+      new CompilationState(from, Some(execPlan))
```

Comment:
We seem to be missing a call to the runtimeSuccessRateMonitor for success.

    runtimeSuccessRateMonitor.newPlanSeen(from.logicalPlan)